mr/src/main/java/org/apache/iceberg/mr/hive/HiveSchemaVisitor.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.hive;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.hadoop.hive.serde2.typeinfo.DecimalTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.ListTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.MapTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.StructTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+
+public class HiveSchemaVisitor {
+  private int id;
+
+  public HiveSchemaVisitor() {
+    id = 0;
+  }
+
+  List<Types.NestedField> visit(List<String> names, List<TypeInfo> typeInfos) {
+    List<Types.NestedField> result = new ArrayList<>(names.size());
+    for (int i = 0; i < names.size(); ++i) {
+      result.add(Types.NestedField.optional(id++, names.get(i), visit(typeInfos.get(i))));
+    }
+
+    return result;
+  }
+
+  Type visit(TypeInfo typeInfo) {
+    switch (typeInfo.getCategory()) {
+      case PRIMITIVE:
+        switch (((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory()) {
+          case FLOAT:
+            return Types.FloatType.get();
+          case DOUBLE:
+            return Types.DoubleType.get();
+          case BOOLEAN:
+            return Types.BooleanType.get();
+          case BYTE:
+          case SHORT:
+          case INT:
+            return Types.IntegerType.get();
+          case LONG:
+            return Types.LongType.get();
+          case BINARY:
+            return Types.BinaryType.get();
+          case STRING:
+          case CHAR:
+          case VARCHAR:
+            return Types.StringType.get();
+          case TIMESTAMP:
+            return Types.TimestampType.withoutZone();
```

Comment:
Returning Types.TimestampType.withoutZone() will use IcebergTimestampObjectInspector.INSTANCE_WITHOUT_ZONE instance to get local time and then throw ClassCastException

Reply:
@qphien: Thanks for reporting this. Could you please help with the concrete repro steps? Based on my understanding the way we store the timestamps in Iceberg should not affect Hive queries.

Thanks,
Peter

Reply:
I think the reported issue is most probably independent from the create table command.
I have tried several different ways to repro anyway, but was not able 😢

Reply:
Sorry for my late reply. I have a iceberg table schema like below:
```
hive (default)> desc complex_iceberg_table_2;
OK
col_name	data_type	comment
id                  	bigint              	from deserializer
user                	string              	from deserializer
asset               	decimal(7,3)        	from deserializer
event_time          	timestamp           	from deserializer
day                 	date                	from deserializer
logs                	array<string>       	from deserializer
location            	map<string,string>  	from deserializer
verbose             	struct<idcomment:string,count:int,elements:array<int>>	from deserializer
```

when i select event_time, hive throws ClassCastException
```
hive (default)> select event_time from complex_iceberg_table_2;
OK
event_time
Failed with exception java.io.IOException:org.apache.hadoop.hive.ql.metadata.HiveException: java.lang.ClassCastException: java.time.OffsetDateTime cannot be cast to java.time.LocalDateTime
```

Reply:
@qphien: Hive2, or Hive3? - Seems like an issue with the timestamp handling itself.
Thanks, Peter

Reply:
@pvary Sorry, it's my mistake. This table iceberg data was produced with this repo commit 3a319f34cc09aa3e5174db6dfb056e0c66946aa5 before. However, i tested hive with the latest commit and got the class cast exception. I reproduce the iceberg data with the latest commit and every works fine.

Thanks for your help.

Reply:
One more question, HiveSchemaVisitor converts hive timestamp to iceberg timestamp, but with https://github.com/pvary/iceberg/blob/ba014580c536aa8d8aec85b0a90e0f48f7d96c55/spark/src/main/java/org/apache/iceberg/spark/SparkTypeToType.java#L148-L149, spark converts java timestamp to  timestamptz. In such case: the iceberg table is created by hive and spark is used to generate iceberg data, will this inconsistency be a problem?

Reply:
Discussed with Ryan and we will move Timestamp related stuff out from this PR to simplify the scope.

There is no good way to interop between Spark and Hive2. AFAIK Spark only supports timestamptz. Hive 2 has a very mixed up timestamp handling dependent on the file format used. Hive 3 has fixed this, and it has 2 different types for timestamp and timestamptz which cleans up this mess and will help us to have a clean mapping.